### ISSUE TITLE
Rule control_statement reports methods with keyword names

### ISSUE THREAD

**Issue by frenetisch-applaudierend** [realm/SwiftLint#1946]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

Swiftlint version: 0.24.0
Installation method: Homebrew

The `control_statement` rule reports methods that have names equal to the control statement keywords. This is an issue if you e.g. use PromiseKit that uses `catch` in one of their method names.

The following triggers the rule and should not in my opinion:

```swift
func test() {
    foo().catch(all: true) {}
}
```

```bash
$ echo "foo().catch(all: true) {}" | swiftlint lint --no-cache --use-stdin --enable-all-rules
Loading configuration from '.swiftlint.yml'
<nopath>:1:7: warning: Control Statement Violation: if,for,while,do,catch statements shouldn't wrap their conditionals or arguments in parentheses. (control_statement)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by Lutzifer** [comment]
Second that
